float64[] path
float64 now_x
float64 now_y
int64 next_dot
float64[] next_p
float64[] passing
